### PR TITLE
various updates

### PR DESCRIPTION
**Pull request by mvoorberg**
support Pg varchar types
fix store package name
fix where clause on Delete & Find sql queries
Log a warning if PK not found
exclude interface{} from the toString

### CODE REVIEW

**Comment by aakash-rajur** [comment]
hi, I'm really glad knowing you find this tool useful!

1. thank you for catch the obvious case of missing `varchar`
2. I've updated the generated files so that we don't see redundant diffs, can you pull in these changes as they're conflicting now?
3. I've also fixed the issue because of mysql behaviour, essentially mysql will not drop tables after a transaction rollback
4. i've also added a local binary generate command in the Taskfile
5. do you think `Makefile` is more accessible than `Taskfile`?

Cheers 🍻 !

File: example/internal/tmdb_mysql/api/get-actor.sql [hunk 1]
```diff
@@ -1,24 +0,0 @@
-select
```
Comment (aakash-rajur) [inline example/internal/tmdb_mysql/api/get-actor.sql]: add all query files back? if we remove it it won't generate queries orm files

File: internal/generate/pg/model.go.tmpl [hunk 2]
```diff
@@ -114,9 +118,9 @@ SELECT
 {{- end }}
 FROM {{ .Table.SchemaName }}.{{ .Table.TableName }}
 WHERE TRUE
-{{- range $selectFields }}
-  AND (CAST(:{{ .Column.ColumnName }} AS {{ .Column.Type | ToUpper }}) IS NULL or {{ .Column.ColumnName }} = :{{ .Column.ColumnName }})
-{{- end }}
+{{- range $pkFields }}
```
Comment (aakash-rajur) [inline internal/generate/pg/model.go.tmpl]: only `pkFields` not necessarily be the case. 
1. what about the scenario where one or more columns have a unique constraint but are not explicitly annotated as primary key?
2. what if we've multiple id columns, one explicitly annotated as primary key and another unique auto incrementing column defined by an arbitrary business use case? what if i want to find or delete by this business id key?

hence `IS NULL` check takes care of a use-case wherein we don't have to provide all the fields

File: sqlxgen
Comment (aakash-rajur) [inline sqlxgen:1]: there pipelines to generate binaries on release and automated homebrew taps, or add it `.gitignore`

**Review by aakash-rajur — CHANGES_REQUESTED** [review]

File: internal/generate/pg/model.go.tmpl (hunk 2 above)
Comment (mvoorberg) [reply internal/generate/pg/model.go.tmpl]: If there are other ways to select a single row, the FindAll should suffice as it already has the ability to build the Where with any provided fields. It's not too difficult to pull row[0] from that result.

File: example/internal/tmdb_mysql/api/get-actor.sql (hunk 1 above)
Comment (mvoorberg) [reply example/internal/tmdb_mysql/api/get-actor.sql]: I don't mind having them all back.

File: sqlxgen
Comment (mvoorberg) [reply sqlxgen:1]: I saw you added the "build.go" task. That's good & easier that figuring it out each time!
I haven't looked into what it looks like to bundle & deploy a release.

File: internal/generate/pg/model.go.tmpl (hunk 2 above)
Comment (mvoorberg) [reply internal/generate/pg/model.go.tmpl]: That said, there is currently a bug in the find query, including a wayward semi-colon before LIMIT. I'll have another PR on it soon.  :)

Templates for queries have not been updated to match - yet.

**Review by mvoorberg — COMMENTED** [review]
I think that covers it for now...
The generated code has changes in it from what looks like an older set of templates...

File: internal/config/sqlx_gen_test.go [hunk 3]
```diff
@@ -534,7 +534,7 @@ configs:
         path: gen/mysql/models
 `
 
-	filepath := path.Join(workDir, "sqlxgen.yaml")
+	filepath := path.Join(workDir, "sqlxgen.yml")
```
Comment (aakash-rajur) [inline internal/config/sqlx_gen_test.go:537]: it should be `yaml`

File: internal/generate/pg/model.go.tmpl (hunk 2 above)
Comment (aakash-rajur) [reply internal/generate/pg/model.go.tmpl]: > If there are other ways to select a single row, the FindAll should suffice as it already has the ability to build the Where with any provided fields. It's not too difficult to pull row[0] from that result.

that was my thought initially, but from the perspective of someone who is consuming this generated code, checking for both `error` and then `length` and `first element` check gets repetitive and tedious very soon which led me to the idea of baking that in!

so this way, you've the option to do both and doesn't make assumptions on how this is consumed. if that makes any sense!

File: internal/generate/mysql/model.go.tmpl [hunk 4]
```diff
@@ -114,9 +114,9 @@ SELECT
 {{- end }}
 FROM {{ .Table.SchemaName }}.{{ .Table.TableName }}
 WHERE TRUE
-{{- range $selectFields }}
-  AND (:{{ .Column.ColumnName }} IS NULL or {{ .Column.ColumnName }} = :{{ .Column.ColumnName }})
-{{- end }}
+{{- range $pkFields }}
+  AND {{ .Column.ColumnName }} = :{{ .Column.ColumnName }}
+{{- end }};
```
Comment (aakash-rajur) [inline internal/generate/mysql/model.go.tmpl]: reason for your semi colon

```
{{- end }};
```

File: internal/config/sqlx_gen_test.go (hunk 3 above)
Comment (mvoorberg) [reply internal/config/sqlx_gen_test.go:537]: The test was failing looking for the yaml file, as it was created as yml. Changing it to yaml let the test pass.

Comment (aakash-rajur) [reply internal/config/sqlx_gen_test.go:537]: yeah i need to take a look at it, but for now i want the test cases to pass

File: internal/generate/mysql/model.go.tmpl (hunk 4 above)
Comment (mvoorberg) [reply internal/generate/mysql/model.go.tmpl]: typo!  It should not be there.

File: internal/generate/mysql/model.go.tmpl [hunk 5]
```diff
@@ -137,7 +137,7 @@ WHERE TRUE
 var {{ .CamelName }}DeleteSql = `
 DELETE FROM {{ .Table.SchemaName }}.{{ .Table.TableName }}
 WHERE TRUE
-{{- range $selectFields }}
+{{- range $pkFields }}
```
Comment (aakash-rajur) [inline internal/generate/mysql/model.go.tmpl]: like i [commented](https://github.com/aakash-rajur/sqlxgen/pull/1#discussion_r1483849789) earlier in the chain above, we're assuming there is always a primary key. I personally have a production use case of multiple id columns co-existing (business use case) and need a way to lookup a single row with this alternate id!

with the existing approach, both scenarios can be accommodated while this accommodates only one!

File: internal/generate/pg/model.go.tmpl [hunk 6]
```diff
@@ -137,7 +141,7 @@ WHERE TRUE
 var {{ .CamelName }}DeleteSql = `
 DELETE FROM {{ .Table.SchemaName }}.{{ .Table.TableName }}
 WHERE TRUE
-{{- range $selectFields }}
+{{- range $pkFields }}
```
Comment (aakash-rajur) [inline internal/generate/pg/model.go.tmpl]: same as above, also this makes `extra` sure that the right row is being deleted!

**Comment by aakash-rajur** [comment]
do let me know if the existing approach is leading to unintended behaviour, thanks!

**Comment by mvoorberg** [comment]
Ok. I'll unwind this change and post a 2nd pr for something to support both
cases.

🙂

**Review by aakash-rajur — APPROVED** [review]

**Comment by aakash-rajur** [comment]
thank you!

**Pull request merged**